packages/universal-components/src/RangeDatePicker/AnimatedDayItemArrow.js

```diff
@@ -34,35 +34,48 @@ const RenderArrows = ({ scale }) => (
 );
 
 export default class AnimatedDayItemArrow extends React.Component<Props> {
-  static scale: Animated.Value = new Animated.Value(0);
+  static scale: Animated.Value;
   static animating: boolean;
+  static numberOfInstances: number;
 
   static handleAnimation() {
-    AnimatedDayItemArrow.scale.setValue(0);
-
-    Animated.sequence([
-      Animated.timing(AnimatedDayItemArrow.scale, {
-        ...SHARED_ANIMATION_CONFIG,
-        toValue: 1,
-      }),
-      Animated.timing(AnimatedDayItemArrow.scale, {
-        ...SHARED_ANIMATION_CONFIG,
-        toValue: 0,
-      }),
-    ]).start(() => {
-      if (AnimatedDayItemArrow.animating) {
+    if (AnimatedDayItemArrow.scale === undefined) {
+      AnimatedDayItemArrow.scale = new Animated.Value(0);
+    }
+    if (
+      AnimatedDayItemArrow.scale &&
+      AnimatedDayItemArrow.scale._value === 0 &&
+      AnimatedDayItemArrow.numberOfInstances !== 0
+    ) {
+      Animated.sequence([
+        Animated.timing(AnimatedDayItemArrow.scale, {
+          ...SHARED_ANIMATION_CONFIG,
+          toValue: 1,
+        }),
+        Animated.timing(AnimatedDayItemArrow.scale, {
+          ...SHARED_ANIMATION_CONFIG,
+          toValue: 0,
+        }),
+      ]).start(() => {
         AnimatedDayItemArrow.handleAnimation();
-      }
-    });
+      });
+    }
   }
 
   componentDidMount() {
-    AnimatedDayItemArrow.animating = true;
-    AnimatedDayItemArrow.handleAnimation();
+    if (!AnimatedDayItemArrow.animating) {
```

Comment:
I think that those conditions and `animating`-`numberOfInstances` combinations starting to be quite complex and slightly hard to read.

Cannot we just simply stop/hide animation when user is touching/dragging picker elements and start it again when he stops. I think it's done in similar way in current Kiwi mobile app.